External/FEXCore/include/FEXCore/IR/IR.h

```diff
@@ -252,76 +251,73 @@ class OrderedNode final {
     void SetUses(uint32_t Uses) { NumUses = Uses; }
 };
 
-static_assert(std::is_trivial<OrderedNode>::value);
-static_assert(std::is_trivially_copyable<OrderedNode>::value);
+static_assert(std::is_trivial_v<OrderedNode>);
+static_assert(std::is_trivially_copyable_v<OrderedNode>);
 static_assert(offsetof(OrderedNode, Header) == 0);
 static_assert(sizeof(OrderedNode) == (sizeof(OrderedNodeHeader) + sizeof(uint32_t)));
 
 struct RegisterClassType final {
   uint32_t Val;
-  operator uint32_t() {
+  constexpr operator uint32_t() const {
     return Val;
   }
-  constexpr bool operator==(RegisterClassType const &rhs) const { return Val == rhs.Val; }
-  constexpr bool operator!=(RegisterClassType const &rhs) const { return !operator==(rhs); }
+  friend constexpr bool operator==(const RegisterClassType&, const RegisterClassType&) = default;
 };
 
 struct CondClassType final {
   uint8_t Val;
-  operator uint8_t() {
+  constexpr operator uint8_t() const {
     return Val;
   }
+  friend constexpr bool operator==(const CondClassType&, const CondClassType&) = default;
 };
 
 struct MemOffsetType final {
   uint8_t Val;
-  operator uint8_t() const {
+  constexpr operator uint8_t() const {
     return Val;
   }
-  int operator ==(const MemOffsetType other) const {
```

Comment:
I don't know if it was intentional to return int from these, and since there was no comment, I just assumed it was a mistake.

Reply:
I believe it was just a mistake.